FGEGraphics/GraphicsHelpers/Models/ShapeGenerators.cs

```diff
@@ -0,0 +1,259 @@
+        List<Vector3> vertices = new((int)vertexCount);
+        List<Vector3> normals = new((int)vertexCount);
+        List<Vector2> texCoords = new((int)vertexCount);
+        uint[] indices = new uint[numIndices];
+        int index = 0;
+        for (uint i = 0; i <= slices; i++)
+        {
+            float phi = i * 2 * MathHelper.Pi / slices;
+            float sinPhi = (float)Math.Sin(phi);
+            float cosPhi = (float)Math.Cos(phi);
+
+            float x = radius * cosPhi;
+            float y = radius * sinPhi;
+            float z = height;
+
+            vertices.Add(new Vector3(x, y, z));
+            normals.Add(new Vector3(x, y, z));
+            texCoords.Add(new Vector2((float)cosPhi, (float)sinPhi));
+
+            if (i < slices)
+            {
+                indices[index++] = vertexCount - 1;
+                indices[index++] = vertexCount - 2 - i;
+                indices[index++] = vertexCount - 2 - i - 1;
+            }
+        }
+        for (uint i = 0; i <= stacks; i++)
+        {
+            float theta = i * MathHelper.Pi / stacks;
+            float cosTheta = (float)Math.Cos(theta);
+            for (uint j = 0; j <= slices; j++)
+            {
+                float phi = j * 2 * MathHelper.Pi / slices;
+                float sinPhi = (float)Math.Sin(phi);
+                float cosPhi = (float)Math.Cos(phi);
+                float x = radius * cosPhi;
+                float y = radius * sinPhi;
+                float z = height * cosTheta;
+                vertices.Add(new Vector3(x, y, z));
+                normals.Add(new Vector3(x, y, 0).Normalized());
+                texCoords.Add(new Vector2(cosPhi, sinPhi));
+                if (i < stacks && j < slices)
+                {
+                    uint currentRow = i * (slices + 1);
+                    uint nextRow = (i + 1) * (slices + 1);
+                    indices[index++] = currentRow + j;
+                    indices[index++] = nextRow + j;
+                    indices[index++] = currentRow + j + 1;
+                    indices[index++] = currentRow + j + 1;
+                    indices[index++] = nextRow + j;
+                    indices[index++] = nextRow + j + 1;
+                }
+            }
+        }
+        for (uint i = 0; i <= slices; i++)
+        {
+            float phi = i * 2 * MathHelper.Pi / slices;
+            float sinPhi = (float)Math.Sin(phi);
+            float cosPhi = (float)Math.Cos(phi);
+            float x = radius * cosPhi;
+            float y = radius * sinPhi;
+            float z = -height;
+            vertices.Add(new Vector3(x, y, z));
+            normals.Add(new Vector3(x, y, z).Normalized());
+            texCoords.Add(new Vector2(cosPhi, sinPhi));
+            if (i < slices)
+            {
+                indices[index++] = 0;
+                indices[index++] = i + 1;
+                indices[index++] = i + 2;
+            }
+        }
+        Array.Reverse(indices);
```

Comment:
Why is there a perpetual reverse on cylinder and torus indices?

Reply:
Remember I had the winding order issue? This fixes that.

Reply:
Just put the values in the other order in the first place instead of reversing it after the fact

Reply:
Ah shit here we go again...lol